sklearn/metrics/tests/test_classification.py

```diff
@@ -483,6 +483,34 @@ def test_matthews_corrcoef_multiclass():
     assert_almost_equal(mcc, 0.)
 
 
+def test_matthews_corrcoef_overflow():  # Bug#9622
+    def mcc_safe(y_true, y_pred):
+        conf_matrix = confusion_matrix(y_true, y_pred)
+        true_pos = conf_matrix[1,1]
+        false_pos = conf_matrix[1,0]
+        false_neg = conf_matrix[0,1]
+        n_points = float(conf_matrix.sum())
```

Comment:
Is this float only for the sake of python 2 division? It's not needed. __future__.division is in effect.

Reply:
This can also be determined as len(y_true)

Reply:
done (the `float` was more along the lines of "code as documentation" - added emphasis _why_ this is "correct").